internal/network/resource_network_integration.go

```diff
@@ -0,0 +1,280 @@
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/mapdefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringdefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	incus "github.com/lxc/incus/v6/client"
+	"github.com/lxc/incus/v6/shared/api"
+	"github.com/lxc/terraform-provider-incus/internal/common"
+	"github.com/lxc/terraform-provider-incus/internal/errors"
+	provider_config "github.com/lxc/terraform-provider-incus/internal/provider-config"
+)
+
+// NetworkIntegrationModel resource data model that matches the schema.
+type NetworkIntegrationModel struct {
+	Name        types.String `tfsdk:"name"`
+	Type        types.String `tfsdk:"type"`
+	Description types.String `tfsdk:"description"`
+	Project     types.String `tfsdk:"project"`
+	Remote      types.String `tfsdk:"remote"`
+	Config      types.Map    `tfsdk:"config"`
+}
+
+// NetworkIntegrationResource represent network integration resource.
+type NetworkIntegrationResource struct {
+	provider *provider_config.IncusProviderConfig
+}
+
+// NewNetworkIntegrationResource returns a new network integration resource.
+func NewNetworkIntegrationResource() resource.Resource {
+	return &NetworkIntegrationResource{}
+}
+
+// Metadata for NetworkIntegrationResource.
+func (r NetworkIntegrationResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = fmt.Sprintf("%s_network_integration", req.ProviderTypeName)
+}
+
+// Schema for NetworkIntegrationResource.
+func (r NetworkIntegrationResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"name": schema.StringAttribute{
+				Required: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+
+			"description": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				Default:  stringdefault.StaticString(""),
+			},
+
+			"remote": schema.StringAttribute{
+				Optional: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+
+			"project": schema.StringAttribute{
+				Optional: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+				},
+			},
+
+			"type": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
```

Comment:
Same question as for the description.

Reply:
The network integration type is both required and not computed on our end.
So this indeed looks wrong, it should be false on both counts.

Reply:
I can't use a default value if the field is not `Computed`:

```
$ TF_ACC=1 go test -v internal/network/resource_network_integration_test.go

- Schema Using Attribute Default For Non-Computed Attribute: Attribute
        "description" must be computed when using default. This is an issue with the
        provider and should be reported to the provider developers.
```

Reply:
I like the idea of this field being optional because it behaves like an "enum", with "ovn" as the default value (currently the only supported network integration type).